File: Book.Core/Helpers/UserParams.cs
namespace Books.Core.Helpers
{
    public class UserParams : QueryParams
    {
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";


export class UserParams {

    gender: string;
    minAge = 18
    maxAge = 99;
    pageNumber = 1
    pageSize = 3;
    orderBy = 'lastActive';

    constructor(user: User) {

        // This is just to match the both values for Gender DDL default selection
        this.gender = user.gender.toLowerCase();
    }
}